api/rest/client/lbclient.go

```diff
@@ -42,112 +58,96 @@ func NewLBClient(cfgs []*Config) (Client, error) {
 		}
 		clients = append(clients, &defaultClient)
 	}
-	return &loadBalancingClient{clients: clients}, nil
+
+	rand.Seed(time.Now().UnixNano())
+
+	return &loadBalancingClient{clients: clients, strategy: strategy, retries: retries, length: len(clients)}, nil
 }
 
 // ID returns information about the cluster Peer.
 func (lc *loadBalancingClient) ID(ctx context.Context) (*api.ID, error) {
-	dc, err := lc.whichClient()
-	if err != nil {
-		return nil, err
+	var err error
+	var id *api.ID
+	for {
+		id, err = lc.whichClient().ID(ctx)
+		if err != nil {
+			break
+		}
 	}
-
-	return dc.ID(ctx)
+	return id, err
```

Comment:
I can run a loop like this in every method, but I am going to look around a bit and see if I can take advantage of similarities in all the functions.